python/ray/serve/tests/test_api.py

```diff
@@ -579,11 +579,71 @@ def g():
     def f():
         return "got f"
 
-    serve.run(f.bind(), name="app1")
+    serve.run(f.bind(), route_prefix="/f", name="app1")
     deployment_info = ray.get(controller._all_running_replicas.remote())
     assert "app1_f" in deployment_info
 
 
+def test_deploy_application_with_same_name(serve_instance):
```

Comment:
What happens if you redeploy an app with same name but different route prefix? Can we test it?

Reply:
good point.